jetbrains-core/src/software/aws/toolkits/jetbrains/services/cloudformation/resources/LambdaResources.kt

```diff
@@ -0,0 +1,50 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.services.cloudformation.resources
+
+import software.amazon.awssdk.services.cloudformation.CloudFormationClient
+import software.amazon.awssdk.services.cloudformation.model.StackResourceSummary
+import software.amazon.awssdk.services.cloudformation.model.StackSummary
+import software.aws.toolkits.jetbrains.core.CachedResource
+import software.aws.toolkits.jetbrains.core.CachedResourceBase
+
+object CloudFormationResources {
+    private val listStacks: CachedResource<List<StackSummary>> by lazy {
+        CloudFormationCachedResource { listStacksPaginator().stackSummaries().toList() }
+    }
+
+    fun listStacks() = listStacks
+
+    fun listStackResources(stackId: String): CachedResource<List<StackResourceSummary>> =
```

Comment:
I think I need to add an explicit id to each cached resources because I'm concerned that the backing cache is using object identity - which obviously means that the cache will never get hit. Will investigate and update.

Reply:
Ah I guessed it was equals based due to you used a data class as the key...I think....